statistics/cmsketch.go

```diff
@@ -281,14 +281,46 @@ func (c *CMSketch) queryHashValue(h1, h2 uint64) uint64 {
 	return uint64(res)
 }
 
+func (c *CMSketch) mergeTopN(lTopN map[uint64][]*TopNMeta, rTopN map[uint64][]*TopNMeta, numTop uint32) {
+	counter := make(map[hack.MutableString]uint64)
+	for _, metas := range lTopN {
+		for _, meta := range metas {
+			counter[hack.String(meta.Data)] += meta.Count
+		}
+	}
+	for _, metas := range rTopN {
+		for _, meta := range metas {
+			counter[hack.String(meta.Data)] += meta.Count
+		}
+	}
+	sorted := make([]uint64, len(counter))
+	for _, cnt := range counter {
+		sorted = append(sorted, cnt)
+	}
+	sort.Slice(sorted, func(i, j int) bool {
+		return sorted[i] > sorted[j]
+	})
+	numTop = mathutil.MinUint32(uint32(len(counter)), numTop)
+	lastTopCnt := sorted[numTop-1]
```

Comment:
if `numTop` is 0, this will panic?  executor/analyze.go #334 set `numTop` is 0

Reply:
No, they don't have top n and won't go into this function.